### PR TITLE
JAMES-2586 - Implement PostgresTableManager

### PR DESCRIPTION
**Pull request by vttranlina**
jira Jira https://issues.apache.org/jira/browse/JAMES-2586

### CODE REVIEW

**Review by quantranhong1999 — APPROVED** [review]

File: backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresTableManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+    Function<PostgresModule, PostgresTableManager> tableManagerFactory = module -> new PostgresTableManager(new PostgresExecutor(connectionFactory.create()
+        .map(c -> c)), module);
+
+    @Test
+    void initializeTableShouldSuccessWhenModuleHasSingleTable() {
+        String tableName = "tableName1";
+
+        PostgresTable table = PostgresTable.name(tableName)
+            .creteTableStep(dslContext -> dslContext.createTable(tableName)
```
Comment (Arsnael) [inline backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresTableManagerTest.java]: Question: why not make sure to use instead `createTableIfNotExist` (as I see in the ticket) so we don't need to catch already exist exception? Same for indexes? 

But I'm ok with this though, better safe than sorry later :)

File: backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTableManager.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    public Mono<Void> initializeTables() {
+        return postgresExecutor.dslContext()
+            .flatMap(dsl -> Flux.fromIterable(module.tables())
+                .flatMap(table -> Mono.from(table.getCreateTableStepFunction().apply(dsl))
+                    .doOnSuccess(any -> LOGGER.info("Table {} created", table.getName()))
+                    .onErrorResume(DataAccessException.class, exception -> {
+                        if (exception.getMessage().contains(String.format("\"%s\" already exists", table.getName()))) {
+                            LOGGER.info("Table {} already exists", table.getName());
```
Comment (Arsnael) [inline backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTableManager.java:47]: Do we really want a log saying the table already exists every time we restart James?

File: backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTableManager.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    public Mono<Void> initializeTableIndexes() {
+        return postgresExecutor.dslContext()
+            .flatMap(dsl -> Flux.fromIterable(module.tableIndexes())
+                .concatMap(index -> Mono.from(index.getCreateIndexStepFunction().apply(dsl))
+                    .doOnSuccess(any -> LOGGER.info("Index {} created", index.getName()))
+                    .onErrorResume(DataAccessException.class, exception -> {
+                        if (exception.getMessage().contains(String.format("\"%s\" already exists", index.getName()))) {
+                            LOGGER.info("Index {} already exists", index.getName());
```
Comment (Arsnael) [inline backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTableManager.java:72]: idem

**Comment by quantranhong1999** [comment]
CI seems unstable:

`com.github.dockerjava.api.exception.NotFoundException: Status 404: {"message":"No such image: apache/tika:2.8.0.0"}`

The image tag is still available though:

image

File: backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresTableManagerTest.java [hunk 4]
```diff
@@ -87,7 +87,7 @@ void initializeTableShouldSuccessWhenModuleHasSingleTable() {
             .createTableStep((dsl, tbn) -> dsl.createTable(tbn)
                 .column("colum1", SQLDataType.UUID.notNull())
                 .column("colum2", SQLDataType.INTEGER)
-                .column("colum3", SQLDataType.VARCHAR(255).notNull()));
+                .column("colum3", SQLDataType.VARCHAR(255).notNull())).noRLS();
```
Comment (Arsnael) [inline backends-common/postgres/src/test/java/org/apache/james/backends/postgres/PostgresTableManagerTest.java:90]: Return to the line for `.noRLS()` statement? Would be easier to read later

Idem for below

**Review by hungphan227 — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**